core/shared/platform/zephyr/zephyr_file.c

```diff
@@ -0,0 +1,990 @@
+    if (!ptr) {
+        return __WASI_EMFILE;
+    }
+    fs_dir_t_init(&ptr->dir);
+
+    handle->fd = index;
+    handle->is_sock = false;
+
+    return __WASI_ESUCCESS;
+}
+
+__wasi_errno_t
+os_renameat(os_file_handle old_handle, const char *old_path,
+            os_file_handle new_handle, const char *new_path)
+{
+    /* `old_handle` need to be the the fd of the file to rename.
+     * `new_handle` will not be used.
+     * paths need to be absolute, no relative path will be accepted.
+     */
+    struct zephyr_fs_desc *ptr = NULL;
+    (void)new_handle;
+
+    char abs_old_path[MAX_FILE_NAME + 1];
+    char abs_new_path[MAX_FILE_NAME + 1];
+
+    snprintf(abs_old_path, MAX_FILE_NAME, "%s/%s", prestat_dir, old_path);
+    snprintf(abs_new_path, MAX_FILE_NAME, "%s/%s", prestat_dir, new_path);
+
+    int rc = fs_rename(abs_old_path, abs_new_path);
+    if (rc < 0) {
+        return convert_errno(-rc);
+    }
+
+    GET_FILE_SYSTEM_DESCRIPTOR(old_handle->fd, ptr);
+
+    ptr->path = strdup(new_path);
+
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_unlinkat(os_file_handle handle, const char *path, bool is_dir)
+{
+    /* `old_handle` need to be the the fd of the file to unlink.
+     * `path` need to be absolute, relative path will not be resolved.
+     */
+    char abs_path[MAX_FILE_NAME + 1];
+    struct zephyr_fs_desc *ptr = NULL;
+
+    snprintf(abs_path, MAX_FILE_NAME, "%s/%s", prestat_dir, path);
+
+    if (is_dir) {
+        return __WASI_ENOTDIR;
+    }
+
+    int rc = fs_unlink(path);
+    if (rc < 0) {
+        return convert_errno(-rc);
+    }
+
+    GET_FILE_SYSTEM_DESCRIPTOR(handle->fd, ptr);
+
+    zephyr_fs_free_obj(ptr);
+
+    return __WASI_ESUCCESS;
+}
+
+__wasi_errno_t
+os_lseek(os_file_handle handle, __wasi_filedelta_t offset,
+         __wasi_whence_t whence, __wasi_filesize_t *new_offset)
+{
+    struct zephyr_fs_desc *ptr = NULL;
+    int zwhence;
+
+    GET_FILE_SYSTEM_DESCRIPTOR(handle->fd, ptr);
+
+    // They have the same value but this is more explicit
+    switch (whence) {
+        case __WASI_WHENCE_SET:
+            zwhence = FS_SEEK_SET;
+            break;
+        case __WASI_WHENCE_CUR:
+            zwhence = FS_SEEK_CUR;
+            break;
+        case __WASI_WHENCE_END:
+            zwhence = FS_SEEK_END;
+            break;
+        default:
+            return __WASI_EINVAL;
+    }
+
+    off_t rc = fs_seek(&ptr->file, (off_t)offset, zwhence);
+    if (rc < 0) {
+        return convert_errno(-rc);
+    }
+
+    *new_offset = (__wasi_filesize_t)rc;
+
+    return __WASI_ESUCCESS;
+}
+
+__wasi_errno_t
+os_fadvise(os_file_handle handle, __wasi_filesize_t offset,
+           __wasi_filesize_t length, __wasi_advice_t advice)
+{
+    return __WASI_ENOSYS;
+}
+
+__wasi_errno_t
+os_isatty(os_file_handle handle)
+{
+    return __WASI_ENOSYS;
+}
+
+os_file_handle
+os_convert_stdin_handle(os_raw_file_handle raw_stdin)
+{
+#ifndef STDIN_FILENO
+#define STDIN_FILENO 0
```

Comment:
Seems that STDIN_FILENO/STDOUT_FILENO/STDERR_FILENO are not used? Is it better to remove them?

Reply:
yes it is. Done